### PR TITLE
sbk map data

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 11 functions (+0.06%) on `us`.

✅ This PR fixes 2 warnings!

File: include/entity.h [hunk 1]
```diff
@@ -291,6 +291,8 @@ typedef struct BombableRockData {
     /* 0x88 */ f32 fragmentFallSpeed[6];
 } BombableRockData; // size = 0xA0
 
+typedef s32 TweesterPath[0];
```
Comment (nanaian) [inline include/entity.h:294]: Should this be 0-size or does it work to do `[]`?

File: include/script_api/macros.h [hunk 2]
```diff
@@ -595,13 +595,23 @@
...
+    EVT_CALL(SetCamPerspective, CAM_DEFAULT, 3, 25, 16, 4096) \
+    EVT_CALL(SetCamBGColor, CAM_DEFAULT, 0, 0, 0) \
+    EVT_CALL(SetCamEnabled, CAM_DEFAULT, TRUE) \
+    EVT_CALL(SetCamLeadPlayer, CAM_DEFAULT, FALSE) \
+
+// similar to previous, but ordered differently
+#define EVT_SETUP_CAMERA_ODD_NO_LEAD() \
     EVT_CALL(SetCamPerspective, CAM_DEFAULT, 3, 25, 16, 4096) \
     EVT_CALL(SetCamBGColor, CAM_DEFAULT, 0, 0, 0) \
     EVT_CALL(SetCamLeadPlayer, CAM_DEFAULT, FALSE) \
```
Comment (nanaian) [inline include/script_api/macros.h:617]: We shouldn't have a trailing `\` on the final lines of these macros I don't think

File: src/world/area_jan/jan_00/jan_00_2_npc.c [hunk 3]
```diff
@@ -0,0 +1,800 @@
...
+    EVT_RETURN
+    EVT_END
+};
+
+
+
+
+
+
+API_CALLABLE(func_80240B4C_B2108C);
```
Comment (nanaian) [inline src/world/area_jan/jan_00/jan_00_2_npc.c:106]: What's with all the spacing?

File: src/world/area_mac/mac_06/mac_06_3.c [hunk 4]
```diff
@@ -28,14 +24,11 @@ NpcSettings N(NpcSettings_JrTroopa) = {
...
-
-s32 D_80243438_867F78 = 0;
-
+f32 D_80243438_867F78 = 0.0f;
 s32 D_8024343C_867F7C = 0;
-
 s32 D_80243440_867F80 = 0;
 
-s32 func_80240E80_8659C0(Evt* script, s32 isInitialCall) {
+API_CALLABLE(N(func_80240E80_8659C0)) {
```
Comment (nanaian) [inline src/world/area_mac/mac_06/mac_06_3.c:31]: Honestly unsure about this macro as it can hide the argument names

**Review by nanaian — CHANGES_REQUESTED** [review]
Few comments/thoughts

File: src/world/area_sbk/sbk.h [hunk 5]
```diff
@@ -2,24 +2,82 @@
...
+extern MapSettings sbk_15_settings;
+extern MapSettings sbk_16_settings;
+extern MapSettings sbk_20_settings;
+extern MapSettings sbk_21_settings;
+extern MapSettings sbk_22_settings;
+extern MapSettings sbk_23_settings;
+extern MapSettings sbk_24_settings;
+extern MapSettings sbk_25_settings;
+extern MapSettings sbk_26_settings;
+//extern MapSettings sbk_30_settings;
```
[the remark this answers is not in the document]
Comment (HailSanta) [reply src/world/area_sbk/sbk.h:53]: I've skipped this one and sbk_02 for now. Both involve a nested include that I haven't quite worked out yet.

File: src/world/area_mac/mac_06/mac_06_3.c (hunk 4 above)
Comment (HailSanta) [reply src/world/area_mac/mac_06/mac_06_3.c:31]: Fair concern. Once youre aware of it, I think the macro makes it easier to visually scan a file and easier to search for all API-callable funcs within a certain set of files. It's self-documenting and enforces uniformity. I've already come across a dozen instances where the args were incorrect -- either spelled wrong, missing isInitialCall, or just void. That said, if we decide against it, we an always find-replace to get rid of it. For now, it saves on typing while the work is being done.

File: src/world/area_jan/jan_00/jan_00_2_npc.c (hunk 3 above)
Comment (HailSanta) [reply src/world/area_jan/jan_00/jan_00_2_npc.c:106]: Mistake.

**Pull request merged**